### PR TITLE
add example mTLS + proxy mode

### PR DESCRIPTION
**Pull request by jxx-gg**
* The example includes 3 binaries, server, target and client.
  * server: it only serves as a proxy, and it doesn't create new sessions.
  * target: it registers at the server, and wait for incoming connections. If a supported incoming connection is received, it dials a pre-specified address.
  * client: it registers at the server, and tries to dial at the [dialTarget, dialTargetType]. Once a tunnel session is established (via server) to the target, it will connect the tunnel session to the local stdIO. This can be used with ssh's ProxyCommand option (see comments in the client.go for an example command).
* The example included two target type SSH and gNMI.
* mTLS support is also added.

### CODE REVIEW

File: cmd/client/client.go [hunk 1]
```diff
@@ -13,137 +13,128 @@
...
-package client
+// Exmaples to use this binary with ssh's ProxyCommand option:
+// TLS:
+//		ssh -o ProxyCommand="client
+//		--tunnel_server_address=localhost:$PORT \
+//		--cert_file=$CERT_FILE \
+//		--dial_target=target1 \
+//		--dial_target_type=SSH" $USER@localhost
+// mTLS:
+// ssh -o ProxyCommand="client
```
Comment (gcsl) [inline cmd/client/client.go:27]: Whitespace to be consistent with above TLS example.

**Review by gcsl — APPROVED** [review]

File: cmd/client/client.go (hunk 1 above)
Comment (jxx-gg) [reply cmd/client/client.go]: Done.

File: cmd/client/client.go [hunk 2]
```diff
@@ -13,137 +13,128 @@
...
-type Config struct {
-	TunnelAddress, DialAddress, ListenAddress, CertFile, Target, TargetType string
-}
+var (
+	tunnelAddress  = flag.String("tunnel_server_address", "", "The address of the tunnel")
+	dialTarget     = flag.String("dial_target", "", "The client uses target to register at the server.")
+	dialTargetType = flag.String("dial_target_type", "", "The type of target protocol, e.g. GNMI or SSH.")
+	certFile       = flag.String("cert_file", "", "The certificate file location")
+	keyFile        = flag.String("key_file", "", "The private key file location")
+	caFile         = flag.String("ca_file", "", "The CA file location (for mTLS). If provided, it will be handled as mTLS")
```
Comment (aaronbee) [inline cmd/client/client.go:60]: I'm not a TLS expert, so I may be getting something wrong here, but these options are a bit confusing:
In the TLS case `-cert_file` is used to specify the CA to verify the server.
In the mTLS case `-ca_file` is used to specify the CA to verify the server, and -cert_file, -key_file are used to specify the client's certificate and key.

What would be more clear is for the -ca_file to always be used to specify the CA to verify the server. You enable mTLS by passing in cert_file, key_file.

Comment (jxx-gg) [reply cmd/client/client.go]: That's a good point. Updated.

File: cmd/client/client.go [hunk 3]
```diff
@@ -13,137 +13,128 @@
...
-type Config struct {
-	TunnelAddress, DialAddress, ListenAddress, CertFile, Target, TargetType string
-}
+var (
+	tunnelAddress  = flag.String("tunnel_server_address", "", "The address of the tunnel")
+	dialTarget     = flag.String("dial_target", "", "The client uses target to register at the server.")
+	dialTargetType = flag.String("dial_target_type", "", "The type of target protocol, e.g. GNMI or SSH.")
+	certFile       = flag.String("cert_file", "", "The certificate file location. If both cert_file and key_file are provided, mTLS will be used.")
+	keyFile        = flag.String("key_file", "", "The private key file location. If both cert_file and key_file are provided, mTLS will be used.")
+	caFile         = flag.String("ca_file", "", "The CA file location (for mTLS)")
```
Comment (aaronbee) [inline cmd/client/client.go:60]: The help message here should be update to not include "(for mTLS)"

Comment (jxx-gg) [reply cmd/client/client.go]: Fixed.

**Pull request merged**